### PR TITLE
Machine Settings initial

### PR DESCRIPTION
**Pull request by usirin**
This is very much a POC to list machines with settings. Improvements can and _should_ be made before merging this (or possibly publishing a new version)

kd-atom-machine-settings

### CODE REVIEW

**Comment by usirin** [comment]
TODO:
- [ ] check with a light ui theme
- [ ] turn on/off should receive related machine instance
- [ ] show logs should receive related machine instance

Not required but good improvement:
- add mount/unmount button for each machine
- add show offline checkbox to list all machines
- better filtering options (filter by each column)
- use `SelectListView` and implement keybindings for each action while this panel is activated
  - turn on/off keybinding
  - show logs keybinding
  - set always on state keybinding
  - possibly mount/unmount keybinding

File: lib/constants.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  1: 'Offline',
+  2: 'Online',
+  3: 'Connected'
+}
+
+export const MachineStatus = {
+  Offline: 1,
+  Online: 2,
+  Connected: 3
+}
```
Comment (sinan) [inline lib/constants.js]: arrays would have been more appropriate.

File: lib/kd-controller.js [hunk 2]
```diff
@@ -127,6 +128,71 @@ export default class KDController extends kd.Object {
     })
   }
 
+  showMachineSettingsView (machines, options = {}) {
```
Comment (sinan) [inline lib/kd-controller.js]: this should replace `kd:machines`

File: lib/views/machine-settings.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            // this.th('id')
+            this.th('Alias')
+            this.th('Team')
+            this.th('Machine Label')
+            this.th('Ip')
+            this.th('Owner')
+            this.th('Provider')
+            this.th('Status')
+            this.th('Created At')
+            this.th('Turn on/off')
```
Comment (sinan) [inline lib/views/machine-settings.js]: need a `Mount` button here as well

File: lib/constants.js (hunk 1 above)
Comment (usirin) [inline lib/constants.js]: What do you mean?

Comment (sinan) [inline lib/constants.js]: One array of `[ null, 'Offline', 'Online', 'Connected']` would have been enough for both uses imo.

Comment (usirin) [inline lib/constants.js]: I especially made this an object to be able to use it like `MachineStatus.Online`, `MachineStatus.Offline` to make it more readable. 

But i guess i can change `MachineStatusText` to be an array.

Comment (usirin) [inline lib/constants.js]: done.

File: lib/views/machine-settings.js (hunk 3 above)
Comment (usirin) [inline lib/views/machine-settings.js]: done.

**Pull request merged**